eokwukwe/ecommerce-shop-api
# #169743926 User Update Address

#### What does this PR do?
Implement address update functionality
 #### Description of Task to be completed?
- [x] Registered user can update their address
 #### How should this be manually tested?
- clone the branch
- Run `npm install` to install dependencies
- Run `npm run dev` to start the development server
- Login to obtain your access token to be sent along in the header as `USER_KEY: Bearer access_token`
- Send a `POST` request to `http://localhost:8000/api/v1/customers/address` with the following credentials:
```
{
  "address_1": "QI 19",
  "city": "madrid",
  "region": "europe",
  "postal_code": "2212121",
  "country": "spain",
  "shipping_region_id": 1
}
```
- Expected output if the credentials are correct: 
```
{
  "customer_id": 1,
  "name": "Lannucci",
  "email": "lannucci@hotmail.com",
  "address_1": "QI 19",
  "address_2": "",
  "city": "madrid",
  "region": "europe",
  "postal_code": "2212121",
  "country": "spain",
  "shipping_region_id": 1,
  "day_phone": "+351323213511235",
  "eve_phone": "+452436143246123",
  "mob_phone": "+351323213511235",
  "credit_card": "XXXXXXXX5100"
}
```

 #### Any background context you want to provide?
Use `2,3 or 4` for the shipping_region_id. Id 1 is the default. If you use any other shipping_region_id outside the specified, users will receive `no region found` response because the region does not yet exist in the database. The region is checked against the database record to make sure it exists. The functionality to add more regions will be added in the future.
 #### What are the relevant pivotal tracker stories?
[#169743926](https://www.pivotaltracker.com/story/show/169743926)
